addon/components/mdc-tab-bar-scroller.js

```diff
@@ -0,0 +1,141 @@
+import { MDCComponent } from '../mixins/mdc-component';
+import layout from '../templates/components/mdc-tab-bar-scroller';
+import getElementProperty from '../utils/get-element-property';
+import getComponentProperty from '../utils/get-component-property';
+import styleComputed from '../utils/style-computed';
+
+const { get, computed, set, run } = Ember;
+const { cssClasses, strings } = MDCTabBarScrollerFoundation;
+
+export default Ember.Component.extend(MDCComponent, {
+  //region Ember Hooks
+  classNames: ['mdc-tab-bar-scroller'],
+  layout,
+  init() {
+    this._super(...arguments);
+    set(this, 'backIndicatorClasses', Ember.A([]));
+    set(this, 'forwardIndicatorClasses', Ember.A([]));
+    set(this, 'mdcScrollFrameStyles', {});
+  },
+  didInsertElement() {
+    this._super(...arguments);
+    set(this, 'scrollFrameElement', getElementProperty(this, 'querySelector', () => null)(strings.FRAME_SELECTOR));
+    set(this, 'tabBarElement', getElementProperty(this, 'querySelector', () => null)(strings.TABS_SELECTOR));
+    set(this, 'forwardIndicatorElement', getElementProperty(this, 'querySelector', () => null)(strings.INDICATOR_FORWARD_SELECTOR));
+    set(this, 'backIndicatorElement', getElementProperty(this, 'querySelector', () => null)(strings.INDICATOR_BACK_SELECTOR));
+  },
+  //endregion
+
+  //region Properties
+  /**
+   * @type {Object}
+   */
+  CLASS_NAMES: cssClasses,
+
+  /**
+   * @property {HTMLElement}
+   */
+  scrollFrameElement: null,
+
+  /**
+   * @property {HTMLElement}
+   */
+  tabBarElement: null,
+
+  /**
+   * @property {HTMLElement}
+   */
+  forwardIndicatorElement: null,
+
+  /**
+   * @property {HTMLElement}
+   */
+  backIndicatorElement: null,
+
+  /**
+   * @type {String[]}
+   */
+  backIndicatorClasses: null,
+
+  /**
+   * @type {String[]}
+   */
+  forwardIndicatorClasses: null,
+
+  /**
+   * @type {Ember.Component}
+   */
+  'tab-bar': null,
+
+  /**
+   * Key value pairs for CSS styles
+   * @type {Object}
+   */
+  mdcScrollFrameStyles: null,
+  //endregion
+
+  //region Methods
+  createFoundation() {
+    return new MDCTabBarScrollerFoundation({
+      addClass: (className) => run(() => get(this, 'mdcClasses').addObject(className)),
+      removeClass: (className) => run(() => get(this, 'mdcClasses').removeObject(className)),
+      eventTargetHasClass: (target, className) => target.classList.contains(className),
+      addClassToForwardIndicator: (className) => run(() => get(this, 'forwardIndicatorClasses').addObject(className)),
+      removeClassFromForwardIndicator: (className) => run(() => get(this, 'forwardIndicatorClasses').removeObject(className)),
+      addClassToBackIndicator: (className) => run(() => get(this, 'backIndicatorClasses').addObject(className)),
+      removeClassFromBackIndicator: (className) => run(() => get(this, 'backIndicatorClasses').removeObject(className)),
+      isRTL: () => getComputedStyle(get(this, 'element')).getPropertyValue('direction') === 'rtl',
```

Comment:
It seems like this might run the risk of `this.element` being missing because the component has been destroyed -- probably safest to change it to something like:

```js
!!this && !get(this, 'isDestroyed') && !!get(this, 'element') && getComputedStyle(get(this, 'element')).getPropertyValue('direction') === 'rtl'
```